src/marten/conf/global_settings/database.cr

```diff
@@ -92,6 +92,35 @@ module Marten
         def initialize(@id : String)
         end
 
+        def from_url(url : String)
+          # URI.parse cant parse 'sqlite://:memory:'
+          if url.starts_with? "sqlite://:memory:"
+            self.backend = "sqlite"
+            self.name = ":memory:"
+            return
+          end
+
+          uri = URI.parse url
+
+          self.backend = uri.scheme
+          self.host = uri.host
+          self.port = uri.port
+          self.user = uri.user
+          self.password = uri.password
+          self.name = uri.scheme == "sqlite" ? uri.host : uri.path[1..]?
```

Comment:
It's worth noting that `sqlite3` is a valid scheme for SQLite databases. We should definitely account for that!